esdl/cube.py

```diff
@@ -91,22 +101,127 @@ def create(base_dir, config=CubeConfig()):
+            'standard_name' : 'time',
+            'units'         : config.time_units,
+            'calendar'      : config.calendar,
+            'bounds'        : 'time_bnds',
+            '_ARRAY_DIMENSIONS' : ['time'],
+        }
+        time_vals = [sn + temporal_res * (i + 0.5) for sn in start_nums for i in range(num_periods_per_year)]
+        # times[-1] = var_time_bnds[-1,0] + (var_time_bnds[-1,1] - var_time_bnds[-1,0]) / 2.
+        # Thus, we keep date of the last time range always at Julian day 364, not in the center of the period.
+        # The time bounds then specify the real extent of the period.
+        # Uncomment the upper line to center it between the upper and lower bound!
+        
+        lon_attrs = {
+            'long_name'     : 'longitude',
+            'standard_name' : 'longitude',
+            'units'         : 'degrees_east',
+            'bounds'        : 'lon_bnds',
+            '_ARRAY_DIMENSIONS' : ['lon'],
+        }
+        lon_bnds_attrs = {'units' : 'degrees_east', '_ARRAY_DIMENSIONS' : ['lon', 'bnds']}
+
+        lat_attrs = {
+            'long_name'     : 'latitude',
+            'standard_name' : 'latitude',
+            'units'         : 'degrees_north',
+            'bounds'        : 'lat_bnds',
+            '_ARRAY_DIMENSIONS' : ['lat'],
+        }
+        
+        lat_bnds_attrs = {'units' : 'degrees_north', '_ARRAY_DIMENSIONS': ['lat', 'bnds']}
+
+        spatial_res = config.spatial_res
+
+        lon0 = config.easting
+        lon_vals = np.zeros(config.grid_width)
+        lon_bnds_vals = np.zeros((config.grid_width,2))
+        for i in range(config.grid_width):
+            lon = lon0 + i * spatial_res
+            lon_vals[i] = lon + 0.5 * spatial_res
+            lon_bnds_vals[i, 0] = lon
+            lon_bnds_vals[i, 1] = lon + spatial_res
+
+        lat0 = config.northing
+        lat_vals = np.zeros(config.grid_height)
+        lat_bnds_vals = np.zeros((config.grid_height,2))
+        for i in range(config.grid_height):
+            lat = lat0 - i * spatial_res
+            lat_vals[i] = lat - 0.5 * spatial_res
+            lat_bnds_vals[i, 0] = lat - spatial_res
+            lat_bnds_vals[i, 1] = lat
+
+        time_dim  = ('time', time_vals)
+        lat_dim  = ('lat', lat_vals)
+        lon_dim  = ('lon', lon_vals)
+        
+        time_bnds_var = xr.DataArray(time_bnds_vals,coords = {'time': time_dim, 'bnds': bndsdim}, dims=['time', 'bnds'])
+        lat_bnds_var = xr.DataArray(lat_bnds_vals,coords = {'lat': lat_dim, 'bnds': bndsdim}, dims=['lat', 'bnds'])
+        lon_bnds_var = xr.DataArray(lon_bnds_vals,coords = {'lon': lon_dim, 'bnds': bndsdim}, dims=['lon', 'bnds'])
+        
+        ds = xr.Dataset({'time_bnds': time_bnds_var,
+                         'lat_bnds' : lat_bnds_var,
+                         'lon_bnds' : lon_bnds_var,
+                        })
+        
+        dsz = ds.to_zarr(base_dir)
+    
+
+        # Write group attributes
```

Comment:
The following is just metadata attribute setting. Why don't you modify the `ds.attrs` directly?